docker-compose.yml

```diff
@@ -41,20 +44,24 @@ services:
       - 8001:8001
       - 8002:8002
     depends_on:
-      - rabbitmq
+      rabbitmq:
+        condition: service_healthy
     volumes:
       - ${LIBRETIME_CONFIG_FILEPATH:-./config.yml}:/etc/libretime/config.yml:ro
       - libretime_playout:/app
     environment:
       LIBRETIME_GENERAL_PUBLIC_URL: http://nginx:8080
+    healthcheck:
+      test: liquidsoap --version
```

Comment:
This isn't a useful check to see if liquidsoap is running because this will return true independently of the actual running process. This verifies that the liquidsoap binary in the container exists and at least somewhat runs, but doesn't check the specific process that libretime-liquidsoap spawns.

LibreTime liquidsoap is a fairly simple script that starts a liquidsoap process and then communicates with it via telnet. A better check would be to verify that the file `radio.liq` exists in the working directory ([created](https://github.com/libretime/libretime/blob/5d2ad4303938055457654e9ab85318c1dccbe2ab/playout/libretime_playout/liquidsoap/main.py#L43-L52) as part of the startup process) and potentially that the liquidsoap process is running/can be communicated with via telnet.

Reply:
Interesting - `liquidsoap --version` seemed like a decent proxy for the container being ready for two reasons:
* With that healthcheck, the liquidsoap container only reached `healthy` state ~30s after creation
* With the dependency between liquidsoap and playout, I didn't see any of these error logs `WARNING  | libretime_playout.liquidsoap.client._client:wait_for_version:62 - could not get version: [Errno 111] Connection refused` anymore, which is the behaviour I'm trying to prevent

That being said, I changed the healthcheck to test that the `radio.liq` file exists and see the same behaviour.

Reply:
I believe there is a 'on started' hook (to comfirm) that we can use to actually use to toggle a ready state. Let it be a file or a http endpoint with a variable, not sure which is cleaner.

This might be useful from playout as well to know when liquidsoap ready is.

Reply:
> I believe there is a 'on started' hook (to comfirm) that we can use to actually use to toggle a ready state

I just checked the liquidsoap docs and found this `on_start` [hook](https://www.liquidsoap.info/doc-1.4.4/reference.html#on_start). I guess that's the one you're referring to. I can see how using it might be the most robust implementation, but I personally don't have the bandwidth to figure out how to do it right now.

Perhaps the current implementation is good enough? It's an improvement over no dependency at all and, as far as I can tell, it solves the immediate issue of playout failing to start because liquidsoap isn't ready.

Curious to hear your thoughts :)

Reply:
This is a good first step. I agree that `on_start` is probably more robust, but can be implemented later down the line